packages/apollo-federation/src/composition/utils.ts

```diff
@@ -91,6 +91,31 @@ export function stripExternalFieldsFromTypeDefs(
         fields,
       };
     },
+    InterfaceTypeExtension(node) {
+      let fields = node.fields;
```

Comment:
It looks like this is identical to the `ObjectTypeExtension`. Could you extract the logic into a function and just do

```
ObjectTypeExtension: removeExternalFieldsFromExtensions,
InterfaceTypeExtension: removeExternalFieldsFromExtensions
```

Reply:
Will do! The type signatures are different so I went with the cheap approach first, but I'll learn me some typescript and make them the same.

Reply:
Done!